Std/Data/BitVec/Basic.lean

```diff
@@ -0,0 +1,200 @@
+/-
+Copyright (c) 2022 by the authors listed in the file AUTHORS and their
+institutional affiliations. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joe Hendrix, Wojciech Nawrocki, Leonardo de Moura
+-/
+namespace Std
+/-!
+We define bitvectors. We choose the `Fin` representation over others for its relative efficiency
+(Lean has special support for `Nat`), alignment with `UIntXY` types which are also represented
+with `Fin`, and the fact that bitwise operations on `Fin` are already defined. Some other possible
+representations are `List Bool`, `{ l : List Bool // l.length = w }`, `Fin w → Bool`.
+
+We also define many bitvector operations from the
+[`QF_BV` logic](https://smtlib.cs.uiowa.edu/logics-all.shtml#QF_BV).
+of SMT-LIBv2.
+
+TODO(Leo): match the interface to what SMT-LIB provides
+-/
+
+/--
+A bitvector of the specified width. This is represented as the underlying `Nat` number
+in both the runtime and the kernel, inheriting all the special support for `Nat`.
+-/
+structure BitVec (w : Nat) where
+  /-- Bitvector is represented as a `Fin`. -/
+  val : Fin (2^w)
+  deriving DecidableEq
+
+namespace BitVec
+
+theorem pow_two_gt_zero (w : Nat) : 2^w > 0 := by
```

Comment:
This should be in `Std.Data.Nat.Init.Lemmas` in the `Nat` namespace:
```lean
theorem pow_two_pos (n : Nat) : 0 < 2 ^ n := Nat.pos_pow_of_pos (by decide)
```

Reply:
Good catch.